tvb/simulator/monitors.py

```diff
@@ -514,6 +521,21 @@ def sample(self, step, state):
         if step % self._period_in_steps == 0:
             time = (step - self._period_in_steps / 2.0) * self.dt
             sample = self._state.copy() / self._period_in_steps
+
+            # By: Adam Li - Needs to Be Checked
+            # add observation noise if available
+            '''
+            1. Should dt be the same as in the source level?
+            2. how should shape be set here for colored noise?
+            '''
+            if self.obsnoise is not None:
+                # configure the noise level
+                if self.obsnoise.ntau > 0.0:
+                    self.obsnoise.configure_coloured(self.dt, sample.shape)
+                else:
+                    self.obsnoise.configure_white(self.dt)
```

Comment:
configuration should be done in the monitor's configure step if possible.  

> Should dt be the same as in the source level?

yes

> how should shape be set here for colored noise?

looks correct to me